cranelift-codegen/meta/src/isa/x86/legalize.rs

```diff
@@ -496,6 +499,54 @@ pub(crate) fn define(shared: &mut SharedDefinitions, x86_instructions: &Instruct
         }
     }
 
+    // SIMD icmp ne
+    let ne = Literal::enumerator_for(&imm.intcc, "ne");
+    for ty in ValueType::all_lane_types().filter(|ty| allowed_simd_type(ty) && ty.is_int()) {
+        let icmp_ = icmp.bind(vector(ty, sse_vector_size));
+        narrow.legalize(
+            def!(c = icmp_(ne, a, b)),
+            vec![def!(x = icmp(eq, a, b)), def!(c = bnot(x))],
+        );
+    }
+
+    // SIMD icmp greater-/less-than
+    let sgt = Literal::enumerator_for(&imm.intcc, "sgt");
+    let ugt = Literal::enumerator_for(&imm.intcc, "ugt");
+    let sge = Literal::enumerator_for(&imm.intcc, "sge");
+    let uge = Literal::enumerator_for(&imm.intcc, "uge");
+    let slt = Literal::enumerator_for(&imm.intcc, "slt");
+    let ult = Literal::enumerator_for(&imm.intcc, "ult");
+    let sle = Literal::enumerator_for(&imm.intcc, "sle");
+    let ule = Literal::enumerator_for(&imm.intcc, "ule");
+    for ty in &[I8, I16, I32] {
+        // greater-than
+        let icmp_ = icmp.bind(vector(*ty, sse_vector_size));
+        narrow.legalize(
+            def!(c = icmp_(ugt, a, b)),
+            vec![def!(x = x86_pmaxu(a, b)), def!(c = icmp(eq, a, x))],
+        );
+        let icmp_ = icmp.bind(vector(*ty, sse_vector_size));
```

Comment:
Why binding this multiple times with the same type?

Reply:
`def_rhs!` (used inside `def!`) uses the `Into` trait of `Instruction/BoundInstruction` to get to an `InstSpec`: https://github.com/CraneStation/cranelift/blob/cabbacbe63477fcc7d1f1e36717d84f08f749a98/cranelift-codegen/meta/src/cdsl/ast.rs#L678-L682

I tried `$inst.clone().into()` but unfortunately neither `Instruction` nor `InstSpec` implements `Clone`. I guess I could add those... other suggestions?